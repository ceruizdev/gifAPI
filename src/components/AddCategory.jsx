import { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const OnInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const OnSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) return;
    onNewCategory(inputValue.trim());
    setInputValue('');
  };

  return (
    <form onSubmit={OnSubmit} aria-label="form">
      <input
        type="text"
        placeholder="Search new gif"
        value={inputValue}
        onChange={OnInputChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
