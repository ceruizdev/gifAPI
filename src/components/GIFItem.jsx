import PropTypes from 'prop-types';
export const GIFItem = ({id, title, url}) => {
  
    return (
        <div className="card">
            <img src = {url} alt={title}/>
            <p>{title}</p>
        </div> 
  )
};

GIFItem.propTypes = {
   id: PropTypes.string.isRequired,
   title: PropTypes.string.isRequired,
   url: PropTypes.string.isRequired
};

