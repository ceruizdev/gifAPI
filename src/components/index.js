export * from './AddCategory';
export * from './GIFGrid';
export * from './GIFItem';