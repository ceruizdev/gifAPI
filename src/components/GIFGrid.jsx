import { useFetchGIFs } from "../hooks/useFetchGIFs";
import { GIFItem } from "./GIFItem";

export const GIFGrid = ({ category }) => {
  const { images, isLoading } = useFetchGIFs(category);

  return (
    <>
      <h3>{category}</h3>
      {
        isLoading && <h3 className="">Cargando...</h3>
      }
      <div className="card-grid">
        {images.map((image) => (
          <GIFItem key={image.id} {...image} />
        ))}
      </div>
    </>
  );
};
