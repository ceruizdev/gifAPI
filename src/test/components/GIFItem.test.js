import { render, screen } from "@testing-library/react";
import { GIFItem } from "../../components";

const title = 'NBA';
const id = '0';
const url = 'http://nba.com/test.jpg';

describe('<GIFItem />', () => {
    test('Match with snapshot', () => {
        const { container } = render(<GIFItem title={title} id={id} url={url}/>);
        expect( container ).toMatchSnapshot();
    });

    test('Show URL and text ALT', () => {
        render(<GIFItem title={title} id={id} url={url}/>);
        const { src, alt} = screen.getByRole('img');
        screen.debug();
        expect( src ).toBe(url);
        expect( alt ).toBe(title);
    });


    test('Show title in the component', () => {
        render(<GIFItem title={title} id={id} url={url}/>);
        expect( screen.getByText(title) ).toBeTruthy();
    });
})