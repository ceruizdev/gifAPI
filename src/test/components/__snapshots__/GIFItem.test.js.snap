// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GIFItem /> Match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="NBA"
      src="http://nba.com/test.jpg"
    />
    <p>
      NBA
    </p>
  </div>
</div>
`;
